Fix build problems related to kroki

Fix build that started failing in https://github.com/ratatui-org/website/pull/491 / https://github.com/ratatui-org/website/pull/490

<details><summary>Details</summary>
<p>

```
2024-03-07T22:12:35.938311Z	Cloning repository...
2024-03-07T22:12:36.883869Z	From https://github.com/ratatui-org/website
2024-03-07T22:12:36.884404Z	 * branch            b2018e83b91310a55074e4a7c66a6c4e65ccbfab -> FETCH_HEAD
2024-03-07T22:12:36.884571Z	
2024-03-07T22:12:38.924462Z	HEAD is now at b2018e8 docs: Remove orphan link (#491)
2024-03-07T22:12:38.924993Z	
2024-03-07T22:12:39.027263Z	
2024-03-07T22:12:39.027916Z	Using v2 root directory strategy
2024-03-07T22:12:39.059986Z	Success: Finished cloning repository files
2024-03-07T22:12:39.90816Z	Restoring from dependencies cache
2024-03-07T22:12:39.927691Z	Restoring from build output cache
2024-03-07T22:12:40.354087Z	Success: Build output restored from build cache.
2024-03-07T22:12:41.871101Z	Success: Dependencies restored from build cache.
2024-03-07T22:12:41.955295Z	Detected the following tools from environment: npm@9.6.7, nodejs@18.17.1
2024-03-07T22:12:41.966621Z	Installing project dependencies: npm clean-install --progress=false
2024-03-07T22:12:48.788826Z	npm WARN deprecated puppeteer@18.2.1: < 21.5.0 is no longer supported
2024-03-07T22:12:58.925645Z	
2024-03-07T22:12:58.92587Z	> ratatui.rs@0.0.1 prepare
2024-03-07T22:12:58.926211Z	> husky install
2024-03-07T22:12:58.926378Z	
2024-03-07T22:12:59.012308Z	install command is deprecated
2024-03-07T22:12:59.067524Z	
2024-03-07T22:12:59.067857Z	added 951 packages, and audited 952 packages in 16s
2024-03-07T22:12:59.068043Z	
2024-03-07T22:12:59.068206Z	306 packages are looking for funding
2024-03-07T22:12:59.068359Z	  run `npm fund` for details
2024-03-07T22:12:59.069876Z	
2024-03-07T22:12:59.070121Z	found 0 vulnerabilities
2024-03-07T22:12:59.123272Z	Executing user command: npm run build
2024-03-07T22:12:59.78617Z	
2024-03-07T22:12:59.786427Z	> ratatui.rs@0.0.1 build
2024-03-07T22:12:59.786586Z	> astro check && astro build
2024-03-07T22:12:59.786701Z	
2024-03-07T22:13:03.141838Z	22:13:03 Types generated 633ms
2024-03-07T22:13:03.150723Z	22:13:03 [check] Getting diagnostics for Astro files in /opt/buildhome/repo...
2024-03-07T22:13:07.27383Z	[96msrc/components/Header.astro[0m:[93m17[0m:[93m3[0m - [93mwarning[0m[90m astro(4000): [0mThis script will be treated as if it has the `is:inline` directive because it contains an attribute. Therefore, features that require processing (e.g. using TypeScript or npm packages in the script) are unavailable.
2024-03-07T22:13:07.280553Z	
2024-03-07T22:13:07.280863Z	See docs for more details: https://docs.astro.build/en/guides/client-side-scripts/#script-processing.
2024-03-07T22:13:07.281025Z	
2024-03-07T22:13:07.281155Z	Add the `is:inline` directive explicitly to silence this hint.
2024-03-07T22:13:07.281269Z	
2024-03-07T22:13:07.281394Z	[7m17[0m   type="text/partytown"
2024-03-07T22:13:07.282203Z	[7m  [0m [93m  ~~~~[0m
2024-03-07T22:13:07.282334Z	
2024-03-07T22:13:07.378385Z	Result (8 files): 
2024-03-07T22:13:07.378718Z	- 0 errors
2024-03-07T22:13:07.378869Z	- 0 warnings
2024-03-07T22:13:07.37901Z	- 1 hint
2024-03-07T22:13:07.436724Z	
2024-03-07T22:13:10.365551Z	22:13:10 Types generated 642ms
2024-03-07T22:13:10.367258Z	22:13:10 [build] output: "static"
2024-03-07T22:13:10.367525Z	22:13:10 [build] directory: /opt/buildhome/repo/dist/
2024-03-07T22:13:10.36767Z	22:13:10 [build] Collecting build info...
2024-03-07T22:13:10.368065Z	22:13:10 [build] ✓ Completed in 874ms.
2024-03-07T22:13:10.36934Z	22:13:10 [build] Building static entrypoints...
2024-03-07T22:13:19.879827Z	22:13:19 [WARN] [vite] [33m[vite:css] Nested CSS was detected, but CSS nesting has not been configured correctly.
2024-03-07T22:13:19.880095Z	Please enable a CSS nesting plugin *before* Tailwind in your configuration.
2024-03-07T22:13:19.880265Z	See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting
2024-03-07T22:13:19.880367Z	29 |  		max-width: 100%;
2024-03-07T22:13:19.880468Z	30 |  
2024-03-07T22:13:19.880608Z	31 |  		&::marker,
2024-03-07T22:13:19.880741Z	   |    ^^^^^^^^^^
2024-03-07T22:13:19.880897Z	32 |  		&::-webkit-details-marker {
2024-03-07T22:13:19.881004Z	   |  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-03-07T22:13:19.881128Z	33 |  			color: var(--sl-color-gray-3);
2024-03-07T22:13:19.881268Z	   |  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-03-07T22:13:19.881409Z	34 |  		}
2024-03-07T22:13:19.881547Z	   |  ^^^[39m
2024-03-07T22:13:19.881703Z	22:13:19 [WARN] [vite] [33m[vite:css] Nested CSS was detected, but CSS nesting has not been configured correctly.
2024-03-07T22:13:19.881836Z	Please enable a CSS nesting plugin *before* Tailwind in your configuration.
2024-03-07T22:13:19.881963Z	See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting
2024-03-07T22:13:19.882103Z	29 |  		max-width: 100%;
2024-03-07T22:13:19.882257Z	30 |  
2024-03-07T22:13:19.882396Z	31 |  		&::marker,
2024-03-07T22:13:19.882529Z	   |    ^^^^^^^^^^
2024-03-07T22:13:19.882669Z	32 |  		&::-webkit-details-marker {
2024-03-07T22:13:19.882797Z	   |  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-03-07T22:13:19.882923Z	33 |  			color: var(--sl-color-gray-3);
2024-03-07T22:13:19.88312Z	   |  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-03-07T22:13:19.883348Z	34 |  		}
2024-03-07T22:13:19.883493Z	   |  ^^^[39m
2024-03-07T22:13:19.901456Z	22:13:19 [WARN] [vite] [33m[vite:css] Nested CSS was detected, but CSS nesting has not been configured correctly.
2024-03-07T22:13:19.90168Z	Please enable a CSS nesting plugin *before* Tailwind in your configuration.
2024-03-07T22:13:19.901837Z	See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting
2024-03-07T22:13:19.90195Z	1  |  starlight-file-tree:where(.astro-p67cqifm){--x-space: 1.5rem;--y-space: 0.125rem;--y-pad: 0;display:block;border:1px solid var(--sl-color-gray-5);padding:1rem;background-color:var(--sl-color-gray-6);font-size:var(--sl-text-xs);font-family:var(--__sl-font-mono);overflow-x:auto}starlight-file-tree:where(.astro-p67cqifm) .directory > details{border:0;padding:0;padding-inline-start:var(--x-space);background:transparent}starlight-file-tree:where(.astro-p67cqifm) .directory > details > summary{margin-inline-start:calc(-1 * var(--x-space));border:0;padding:var(--y-pad) 0.625rem;font-weight:normal;color:var(--sl-color-white);max-width:100%;&::marker,&::-webkit-details-marker{color:var(--sl-color-gray-3)}&:hover,&:hover .tree-icon{cursor:pointer;color:var(--sl-color-text-accent);fill:var(--sl-color-text-accent);&~ul{border-color:var(--sl-color-gray-4)}}&:hover .highlight .tree-icon{fill:var(--sl-color-text-invert)}}starlight-file-tree:where(.astro-p67cqifm) ul{margin-inline-start:0.5rem;border-inline-start:1px solid var(--sl-color-gray-5);padding:0;padding-inline-start:0.125rem;list-style:none}starlight-file-tree:where(.astro-p67cqifm)>ul{margin:0;border:0;padding:0}starlight-file-tree:where(.astro-p67cqifm) li{margin:var(--y-space) 0;padding:var(--y-pad) 0}starlight-file-tree:where(.astro-p67cqifm) .file{margin-inline-start:calc(var(--x-space) - 0.125rem);color:var(--sl-color-white)}starlight-file-tree:where(.astro-p67cqifm) .tree-entry{display:inline-flex;align-items:flex-start;flex-wrap:wrap;max-width:calc(100% - 1rem);@media (min-width: 30em){&{flex-wrap:nowrap}}&>:first-child{flex-shrink:0}}starlight-file-tree:where(.astro-p67cqifm) .empty{color:var(--sl-color-gray-3);padding-inline-start:0.375rem}starlight-file-tree:where(.astro-p67cqifm) .comment{color:var(--sl-color-gray-3);padding-inline-start:1.625rem;max-width:24rem;min-width:12rem}starlight-file-tree:where(.astro-p67cqifm) .highlight{display:inline-block;border-radius:0.25rem;padding-inline-end:0.5rem;color:var(--sl-color-text-invert);background-color:var(--sl-color-text-accent)}starlight-file-tree:where(.astro-p67cqifm) svg{display:inline;fill:var(--sl-color-gray-3);vertical-align:middle;margin-inline:0.25rem 0.375rem;width:0.875rem;height:0.875rem}starlight-file-tree:where(.astro-p67cqifm) .highlight svg.tree-icon{fill:var(--sl-color-text-invert)}
2024-03-07T22:13:19.902138Z	   |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[39m
2024-03-07T22:13:19.902336Z	22:13:19 [WARN] [vite] [33m[vite:css] Nested CSS was detected, but CSS nesting has not been configured correctly.
2024-03-07T22:13:19.902513Z	Please enable a CSS nesting plugin *before* Tailwind in your configuration.
2024-03-07T22:13:19.902845Z	See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting
2024-03-07T22:13:19.903069Z	1  |  starlight-file-tree:where(.astro-p67cqifm){--x-space: 1.5rem;--y-space: 0.125rem;--y-pad: 0;display:block;border:1px solid var(--sl-color-gray-5);padding:1rem;background-color:var(--sl-color-gray-6);font-size:var(--sl-text-xs);font-family:var(--__sl-font-mono);overflow-x:auto}starlight-file-tree:where(.astro-p67cqifm) .directory > details{border:0;padding:0;padding-inline-start:var(--x-space);background:transparent}starlight-file-tree:where(.astro-p67cqifm) .directory > details > summary{margin-inline-start:calc(-1 * var(--x-space));border:0;padding:var(--y-pad) 0.625rem;font-weight:normal;color:var(--sl-color-white);max-width:100%;&::marker,&::-webkit-details-marker{color:var(--sl-color-gray-3)}&:hover,&:hover .tree-icon{cursor:pointer;color:var(--sl-color-text-accent);fill:var(--sl-color-text-accent);&~ul{border-color:var(--sl-color-gray-4)}}&:hover .highlight .tree-icon{fill:var(--sl-color-text-invert)}}starlight-file-tree:where(.astro-p67cqifm) ul{margin-inline-start:0.5rem;border-inline-start:1px solid var(--sl-color-gray-5);padding:0;padding-inline-start:0.125rem;list-style:none}starlight-file-tree:where(.astro-p67cqifm)>ul{margin:0;border:0;padding:0}starlight-file-tree:where(.astro-p67cqifm) li{margin:var(--y-space) 0;padding:var(--y-pad) 0}starlight-file-tree:where(.astro-p67cqifm) .file{margin-inline-start:calc(var(--x-space) - 0.125rem);color:var(--sl-color-white)}starlight-file-tree:where(.astro-p67cqifm) .tree-entry{display:inline-flex;align-items:flex-start;flex-wrap:wrap;max-width:calc(100% - 1rem);@media (min-width: 30em){&{flex-wrap:nowrap}}&>:first-child{flex-shrink:0}}starlight-file-tree:where(.astro-p67cqifm) .empty{color:var(--sl-color-gray-3);padding-inline-start:0.375rem}starlight-file-tree:where(.astro-p67cqifm) .comment{color:var(--sl-color-gray-3);padding-inline-start:1.625rem;max-width:24rem;min-width:12rem}starlight-file-tree:where(.astro-p67cqifm) .highlight{display:inline-block;border-radius:0.25rem;padding-inline-end:0.5rem;color:var(--sl-color-text-invert);background-color:var(--sl-color-text-accent)}starlight-file-tree:where(.astro-p67cqifm) svg{display:inline;fill:var(--sl-color-gray-3);vertical-align:middle;margin-inline:0.25rem 0.375rem;width:0.875rem;height:0.875rem}starlight-file-tree:where(.astro-p67cqifm) .highlight svg.tree-icon{fill:var(--sl-color-text-invert)}
2024-03-07T22:13:19.903368Z	   |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[39m
2024-03-07T22:14:16.136725Z	TypeError: Failed to parse Markdown file "/opt/buildhome/repo/src/content/docs/concepts/application-patterns/the-elm-architecture.md":
2024-03-07T22:14:16.137171Z	Cannot read properties of null (reading 'attrs')
2024-03-07T22:14:16.137358Z	    at getTokenAttr (file:///opt/buildhome/repo/node_modules/parse5/dist/common/token.js:14:24)
2024-03-07T22:14:16.13768Z	    at Tokenizer._leaveAttrName (file:///opt/buildhome/repo/node_modules/parse5/dist/tokenizer/index.js:353:13)
2024-03-07T22:14:16.137854Z	    at Tokenizer._stateAttributeName (file:///opt/buildhome/repo/node_modules/parse5/dist/tokenizer/index.js:1578:22)
2024-03-07T22:14:16.137989Z	    at Tokenizer._callState (file:///opt/buildhome/repo/node_modules/parse5/dist/tokenizer/index.js:686:22)
2024-03-07T22:14:16.138094Z	    at Tokenizer._runParsingLoop (file:///opt/buildhome/repo/node_modules/parse5/dist/tokenizer/index.js:223:22)
2024-03-07T22:14:16.138206Z	    at Tokenizer.write (file:///opt/buildhome/repo/node_modules/parse5/dist/tokenizer/index.js:248:14)
2024-03-07T22:14:16.138302Z	    at handleRaw (file:///opt/buildhome/repo/node_modules/hast-util-raw/lib/index.js:349:26)
2024-03-07T22:14:16.138406Z	    at one (file:///opt/buildhome/repo/node_modules/zwitch/index.js:108:17)
2024-03-07T22:14:16.138501Z	    at Object.handle (file:///opt/buildhome/repo/node_modules/hast-util-raw/lib/index.js:100:7)
2024-03-07T22:14:16.138597Z	    at all (file:///opt/buildhome/repo/node_modules/hast-util-raw/lib/index.js:156:13)
2024-03-07T22:14:16.138699Z	    at root (file:///opt/buildhome/repo/node_modules/hast-util-raw/lib/index.js:172:3)
2024-03-07T22:14:16.138794Z	    at one (file:///opt/buildhome/repo/node_modules/zwitch/index.js:108:17)
2024-03-07T22:14:16.138893Z	    at raw (file:///opt/buildhome/repo/node_modules/hast-util-raw/lib/index.js:106:3)
2024-03-07T22:14:16.139008Z	    at file:///opt/buildhome/repo/node_modules/rehype-raw/lib/index.js:33:41
2024-03-07T22:14:16.139118Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:160:27)
2024-03-07T22:14:16.139235Z	    at next (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:86:23)
2024-03-07T22:14:16.139337Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.139431Z	    at then (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:204:5)
2024-03-07T22:14:16.139525Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.139744Z	    at next (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:86:23)
2024-03-07T22:14:16.139925Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.140085Z	    at then (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:204:5)
2024-03-07T22:14:16.140253Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.140413Z	    at next (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:86:23)
2024-03-07T22:14:16.140575Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.140766Z	    at then (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:204:5)
2024-03-07T22:14:16.14093Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.141085Z	    at next (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:86:23)
2024-03-07T22:14:16.14126Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.141416Z	    at then (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:204:5)
2024-03-07T22:14:16.141585Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.141742Z	    at next (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:86:23)
2024-03-07T22:14:16.141898Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.142045Z	    at then (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:204:5)
2024-03-07T22:14:16.142219Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.142374Z	    at next (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:86:23)
2024-03-07T22:14:16.142517Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.142657Z	    at then (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:204:5)
2024-03-07T22:14:16.142826Z	    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-03-07T22:14:16.142979Z	22:14:16 [ERROR] [vite] [31mx[39m Build failed in 1m 6s
2024-03-07T22:14:16.716617Z	[astro:markdown] Could not load /opt/buildhome/repo/src/content/docs/concepts/application-patterns/the-elm-architecture.md (imported by src/content/docs/concepts/application-patterns/the-elm-architecture.md?astroPropagatedAssets): Failed to parse Markdown file "/opt/buildhome/repo/src/content/docs/concepts/application-patterns/the-elm-architecture.md":
2024-03-07T22:14:16.716875Z	Cannot read properties of null (reading 'attrs')
2024-03-07T22:14:16.717507Z	  Stack trace:
2024-03-07T22:14:16.717773Z	    at getTokenAttr (file:///opt/buildhome/repo/node_modules/parse5/dist/common/token.js:14:24)
2024-03-07T22:14:16.718308Z	    at Tokenizer._stateAttributeName (file:///opt/buildhome/repo/node_modules/parse5/dist/tokenizer/index.js:1578:22)
2024-03-07T22:14:16.71844Z	    at Tokenizer._runParsingLoop (file:///opt/buildhome/repo/node_modules/parse5/dist/tokenizer/index.js:223:22)
2024-03-07T22:14:16.718714Z	    at handleRaw (file:///opt/buildhome/repo/node_modules/hast-util-raw/lib/index.js:349:26)
2024-03-07T22:14:16.718884Z	    at Object.handle (file:///opt/buildhome/repo/node_modules/hast-util-raw/lib/index.js:100:7)
2024-03-07T22:14:16.719004Z	    at root (file:///opt/buildhome/repo/node_modules/hast-util-raw/lib/index.js:172:3)
2024-03-07T22:14:16.719143Z	    at raw (file:///opt/buildhome/repo/node_modules/hast-util-raw/lib/index.js:106:3)
2024-03-07T22:14:16.719289Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:160:27)
2024-03-07T22:14:16.71942Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.719552Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.719699Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.719904Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.720066Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.720257Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.720397Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.720532Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.720662Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.720792Z	    at wrapped (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:181:9)
2024-03-07T22:14:16.720939Z	    at done (file:///opt/buildhome/repo/node_modules/trough/lib/index.js:194:7)
2024-03-07T22:14:16.721088Z	    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-03-07T22:14:16.757512Z	Failed: Error while executing user command. Exited with error code: 1
2024-03-07T22:14:16.766083Z	Failed: build command exited with code: 1
2024-03-07T22:14:17.710172Z	Failed: error occurred while running build command
```

</p>
</details> 

Locally, the build fails, and moving to a previously known good commit also fails. Deleting most of the TEA article makes this succeed, but not in an obvious consistent manner that indicates the cause of the failure.

I'm not sure why moving between commits that previously worked fine (and re-running npm install) does not fix this.

--- a/astro.config.mjs
+++ b/astro.config.mjs
@@ -4,7 +4,7 @@ import tailwind from "@astrojs/tailwind";
 import { pluginCollapsibleSections } from "@expressive-code/plugin-collapsible-sections";
 import { defineConfig } from "astro/config";
 import emoji from "remark-emoji";
-import { remarkKroki } from "remark-kroki";
+import remarkMermaid from "remark-mermaidjs";
 import remarkYoutube from "remark-youtube";
 import starlightLinksValidator from "starlight-links-validator";
 import { collapsibleFrames } from "/src/plugins/collapsible-frames";
@@ -25,18 +25,7 @@ export default defineConfig({
     prefetchAll: true,
   },
   markdown: {
-    remarkPlugins: [
-      remarkIncludeCode,
-      emoji,
-      [
-        remarkKroki,
-        {
-          server: "https://kroki.io/",
-          output: "inline-svg",
-        },
-      ],
-      remarkYoutube,
-    ],
+    remarkPlugins: [remarkIncludeCode, emoji, remarkMermaid, remarkYoutube],
   },
   integrations: [
     starlight({

--- a/package.json
+++ b/package.json
@@ -11,7 +11,8 @@
     "prepare": "husky install",
     "_format": "prettier . --plugin=prettier-plugin-astro --plugin=prettier-plugin-svelte --plugin=prettier-plugin-organize-imports --plugin=prettier-plugin-tailwindcss",
     "format": "npm run _format -- --write",
-    "format:check": "npm run _format -- --check"
+    "format:check": "npm run _format -- --check",
+    "preinstall": "npx playwright install"
   },
   "dependencies": {
     "@astrojs/check": "^0.5.4",
@@ -24,7 +25,6 @@
     "astro-diagram": "^0.7.0",
     "hast-util-select": "^6.0.2",
     "hastscript": "^9.0.0",
-    "mermaid": "^10.8.0",
     "sharp": "^0.33.2",
     "starlight-links-validator": "^0.5.3",
     "tailwindcss": "^3.4.1",
@@ -34,6 +34,7 @@
   "devDependencies": {
     "husky": "^9.0.11",
     "lint-staged": "^15.2.2",
+    "playwright": "^1.42.1",
     "prettier": "3.2.5",
     "prettier-plugin-astro": "^0.13.0",
     "prettier-plugin-organize-imports": "^3.2.4",
@@ -42,6 +43,7 @@
     "remark": "^15.0.1",
     "remark-emoji": "^4.0.1",
     "remark-kroki": "^0.3.2",
+    "remark-mermaidjs": "^6.0.0",
     "remark-youtube": "^1.3.1",
     "tailwindcss-animate": "^1.0.7",
     "unist-util-visit": "^5.0.0"

--- a/src/content/docs/concepts/application-patterns/the-elm-architecture.md
+++ b/src/content/docs/concepts/application-patterns/the-elm-architecture.md
@@ -26,7 +26,7 @@ At its core, TEA is split into three main components:
 - **View**: This function is responsible for displaying your model to the user. In Elm, it produces
   HTML. In our case, it'll produce terminal UI elements.
 
-```kroki type=mermaid
+```mermaid
 sequenceDiagram
 participant User
 participant TUI Application
@@ -147,7 +147,7 @@ series of interconnected events.
 
 Here's a state transition diagram of the counter example from above:
 
-```kroki type=mermaid
+```mermaid
 stateDiagram-v2
     state Model {
         counter : counter = 0

--- a/src/content/docs/concepts/backends/comparison.md
+++ b/src/content/docs/concepts/backends/comparison.md
@@ -19,7 +19,7 @@ colors and other features. As of now, `ratatui` supports three backends:
 Selecting a backend does influence your project's structure, but the core functionalities remain
 consistent across all options. Here's a flowchart that can help you make your decision.
 
-```kroki type=mermaid
+```mermaid
 graph TD;
     Q1[Is the TUI only for Wezterm users?]
     Q2[Is Windows compatibility important?]

--- a/src/content/docs/concepts/layout.md
+++ b/src/content/docs/concepts/layout.md
@@ -6,7 +6,7 @@ The coordinate system in Ratatui runs left to right, top to bottom, with the ori
 top left corner of the terminal. The x and y coordinates are represented by u16 values and are
 generally listed in that order in most places.
 
-```kroki type=svgbob
+```svgbob
 "(0,0)"-------------> x "(columns)"
 
    |
@@ -76,7 +76,7 @@ frame.render_widget(
 
 This might look something like:
 
-```kroki type=svgbob
+```svgbob
 ┌───────────────────────────────────┐
 │Top                                │
 │                                   │
@@ -143,7 +143,7 @@ frame.render_widget(
     inner_layout[1]);
 ```
 
-```kroki type=svgbob
+```svgbob
 ┌───────────────────────────────────┐
 │outer 0                            │
 │                                   │

--- a/src/content/docs/concepts/rendering/under-the-hood.md
+++ b/src/content/docs/concepts/rendering/under-the-hood.md
@@ -98,7 +98,7 @@ example above, widgets interact with these `Cell`s using `Buffer` methods.
 
 Here's a visual representation of a `Buffer` that is 12 `Cell`s wide and 4 `Cell`s tall.
 
-```kroki type=svgbob
+```svgbob
         0     1     2     3     4     5     6     7     8     9    10    11
      ┌─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┐
    0 │  H  │  e  │  l  │  l  │  o  │     │  W  │  o  │  r  │  l  │  d  │  !  │
@@ -192,7 +192,7 @@ that is passed to the closure provided by the application. The closure draws eac
 buffer by calling the `Frame::render_widget`, which in turn calls each widget's render method.
 Finally, Ratatui writes the contents of the buffer to the terminal.
 
-```kroki type=mermaid
+```mermaid
 sequenceDiagram
     participant A as App
     participant C as Crossterm

--- a/src/content/docs/developer-guide/website.mdx
+++ b/src/content/docs/developer-guide/website.mdx
@@ -101,7 +101,7 @@ Draw diagrams with [`svgbob`]:
 [`svgbob`]: https://github.com/ivanceras/svgbob
 
 ````markdown
-```kroki type=svgbob
+```svgbob
  ,-------------.
  |Get Key Event|
  `-----+-------'
@@ -120,7 +120,7 @@ Draw diagrams with [`svgbob`]:
 
 This renders as:
 
-```kroki type=svgbob
+```svgbob
  ,-------------.
  |Get Key Event|
  `-----+-------'
@@ -143,7 +143,7 @@ Draw diagrams with [`mermaidjs`]:
 [`mermaidjs`]: https://mermaid.js.org/
 
 ````markdown
-```kroki type=mermaid
+```mermaid
 sequenceDiagram
 participant User
 participant TUI Application
@@ -157,7 +157,7 @@ TUI Application-->>User: Display UI
 
 This renders as:
 
-```kroki type=mermaid
+```mermaid
 sequenceDiagram
 participant User
 participant TUI Application

--- a/src/content/docs/faq.md
+++ b/src/content/docs/faq.md
@@ -48,7 +48,7 @@ If the answer is not much, maybe it is simpler to not use `async` and avoiding `
 
 Another way to think about it is, do you think your app would work better with 1 thread like this?
 
-```kroki type=svgbob
+```svgbob
  ,-------------.
  |Get Key Event|
  `-----+-------'
@@ -73,7 +73,7 @@ up on the work done by those tasks in the next iteration.
 
 Or would it work with 3 threads / `tokio` tasks like this:
 
-```kroki type=svgbob
+```svgbob
     Render Thread       ┊         Event Thread             ┊     Main Thread
                         ┊                                  ┊
                         ┊      ,------------------.        ┊
@@ -105,7 +105,7 @@ The former can be done without any `async` code and the latter is the approach s
 
 The `simple-async` uses this architecture instead with tokio:
 
-```kroki type=svgbob
+```svgbob
        Event Thread             ┊     Main Thread
                                 ┊
     ,------------------.        ┊

--- a/src/content/docs/highlights/v0.24.md
+++ b/src/content/docs/highlights/v0.24.md
@@ -120,7 +120,7 @@ fixed and we are able to render charts smaller than 3 lines!
 
 For example, here is how `BarChart` is rendered and resized from a single line to 4 lines in order:
 
-```kroki type=svgbob
+```svgbob
   ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
 
 
@@ -139,7 +139,7 @@ a b c d e f g h i
 
 If you set `bar_width(2)` for 3 lines, then it is rendered as:
 
-```kroki type=svgbob
+```svgbob
           ▂ ▄ ▆ █
   ▂ ▄ ▆ 4 5 6 7 8
 a b c d e f g h i
@@ -170,7 +170,7 @@ Block::default()
 
 When rendered:
 
-```kroki type=svgbob
+```svgbob
 1TTTTTTTTTTTTT2
 L             R
 3BBBBBBBBBBBBB4

--- a/src/content/docs/templates/component/app-rs.md
+++ b/src/content/docs/templates/component/app-rs.md
@@ -7,7 +7,7 @@ we should discuss the process of a TUI.
 
 Most TUIs are single process, single threaded applications.
 
-```kroki type=svgbob
+```svgbob
  ,-------------.
  |Get Key Event|
  `-----+-------'
@@ -285,7 +285,7 @@ impl App {
 
 Now our final architecture would look like this:
 
-```kroki type=svgbob
+```svgbob
     Render Thread               Event Thread                  Main Thread
 
                              ,------------------.
@@ -314,7 +314,7 @@ You can change around when "thread" or "task" does what in your application if y
 It is up to you to decide is this pattern is worth it. In this template, we are going to keep things
 a little simpler. We are going to use just one thread or task to handle all the `Event`s.
 
-```kroki type=svgbob
+```svgbob
        Event Thread                            Main Thread
 
     ,------------------.

--- a/src/content/docs/tutorials/counter-app/multiple-functions.md
+++ b/src/content/docs/tutorials/counter-app/multiple-functions.md
@@ -313,7 +313,7 @@ fn main() -> Result<()> {
 
 Here's a flow chart representation of the various steps in the program:
 
-```kroki type=mermaid
+```mermaid
 graph TD
     MainRun[Main: Run];
     CheckEvent[Main: Poll KeyPress];

--- a/src/content/docs/tutorials/json-editor/ui-main.md
+++ b/src/content/docs/tutorials/json-editor/ui-main.md
@@ -80,7 +80,7 @@ Finally, we are going to create our first nested layout. Because the `Layout.spl
 requires a `Rect`, and not a `Frame`, we can pass one of our chunks from the previous layout as the
 space for the new layout. If you remember the bottom most section from the above graphic:
 
-```kroki type=svgbob
+```svgbob
 +----------------------------------+ Constraint::Length  == 3
 |       This section should        |
 |     always be 3 lines tall       |
@@ -96,7 +96,7 @@ We will create a new layout in this space by passing it (`chunks[2]`) as the par
 
 This code is the visual equivalent of this:
 
-```kroki type=svgbob
+```svgbob
 +---------------------------------+ Constraint::Length  == 3
 |                |                |
 | Length == 50%  | Length == 50%  |

--- a/src/content/docs/tutorials/json-editor/ui.md
+++ b/src/content/docs/tutorials/json-editor/ui.md
@@ -49,7 +49,7 @@ In the example above, you can read the instructions aloud like this:
 
 For those visual learners, I have the following graphic:
 
-```kroki type=svgbob
+```svgbob
 +--------------------------------------------+
 |          Top segment always                | Constraint::Length  == 3
 |          remains 3 lines                   |

